src/Test/TestCombatMods.java

```diff
@@ -158,4 +163,84 @@ private boolean testRetribution()
 
     return testPassed;
   }
+
+  private boolean testDamageDealtToIncomeConverter()
+  {
+    boolean testPassed = true; // Assume nothin's busted
+    DamageDealtToIncomeConverter ddtic = DamageDealtToIncomeConverter.instance(testGame, DamageDealtToIncomeConverter.class);
+
+    Unit aa = addUnit(testMap, cinder, UnitModel.SURFACE_TO_AIR, 7, 3);
+    // juice aa to be extra angry and get overkills
+    aa.alterHP(10, true);
+    aa.addUnitModifier(new CODamageModifier(42));
+
+    int currentFundsReturn = 0;
+    // Get lots of free money, and make sure we get the right amount
+    for( int i = 0; i < 100; ++i )
+    {
+      currentFundsReturn += i;
+      ddtic.startTracking(cinder, i);
+      cinder.money = 0;
+
+      // give aa ammo
+      aa.isTurnOver = false;
+      ResupplyEvent event = new ResupplyEvent(null, aa);
+      event.performEvent( testGame.gameMap );
+      GameEventListener.publishEvent(event, testGame);
+
+      // Add the victim, and yeet him
+      addUnit(testMap, venge, UnitModel.TROOP, 7, 4);
+      performGameAction(new BattleLifecycle.BattleAction(testMap, aa, Utils.findShortestPath(aa, 7, 3, testMap), 7, 4), testGame);
+
+      testPassed &= validate(cinder.money == currentFundsReturn * 1000, "    Expected to make "+currentFundsReturn*1000+", but got "+cinder.money+" instead.");
+    }
+    // Remove those amounts, and make sure that works
+    for( int i = 0; i < 55; ++i )
+    {
+      currentFundsReturn -= i;
+      ddtic.stopTracking(cinder, i);
+      ddtic.stopTracking(cinder, i); // Remove twice, just to be sure
```

Comment:
What? Why? This is literally inside a test case. If you want to be sure, put in a test that will fail if it isn't doing what it should.

Reply:
Also, if you tell it to stop tracking, shouldn't it fail to increase Cinder's warchest?

Reply:
That's the intent *of* the test case - I'm making sure double-removal doesn't break my assumptions.
This is the test that would fail.

`stopTracking()` works on one particular value at a time - there's probably a better name for it, but the semantics are "Cancel giving me money at this particular exchange rate, assuming you were giving me money in the first place"

Reply:
Ah. Well, the comment seems to say "We'd better remove this value twice, since once may not be enough to do the job". Which implies the test case is working around known deficiencies in the implementation.

Sometimes your comments leave me more confused than if there wasn't one at all.